docs/advanced-usage.md

```diff
@@ -2,13 +2,31 @@
 
 FakeItEasy exposes a few APIs that aren't commonly needed, but can be useful in certain scenarios.
 
-## Clearing the configuration of a fake
+## Resetting a fake to its initial state
 
-To "unconfigure" a fake so it has default behavior, discard the fake and create
-a new one. If this is not feasible, for example because you wish to remove the
-configuration in the middle of a test and the system under test already holds
-the fake, see
-[Changing behavior between calls](changing-behavior-between-calls.md).
+In most cases, you should just discard the fake after use and create a new one; in other words,
+don't reuse a fake between tests if you can avoid it.
+
+However, in some situations this might not be practical. To actually reset a fake to its initial
+state, use the `Fake.Reset` method:
+
+```csharp
+Fake.Reset(fake);
+```
+
+This method resets all changes made to the fake after it was created. This includes:
+* call configurations

```

Comment:
It's hard to add notes to unmodified code, so I'm noting here.

Down in "Manipulating a fake's call rules" we should probably explain about `IStatefulFakeObjectCallRule`.